### DISCUSSION TITLE
React开发是否考虑将style的定义改为CSSProperties。

### DISCUSSION THREAD

**Discussion by Tenodec** [NervJS/taro#12917] [Q & A] [answered]
### 这个特性解决了什么问题？
既然使用ts开发，但是style又是string，隔壁remax支持CSSProperties（虽然因为优化原因不在维护），现在要迁移到taro，结果sytle的定义不一样

### 这个 API 长什么样？
```
style?:string改为
style?:CSSProperties
```

**Comment by AdvancedCat** [comment, marked as answer]
应该是支持 CSSProperites 的呀

https://github.com/NervJS/taro/blob/178eb37c7a028fbe7b626a59ecb610a1871792db/packages/taro-components/types/common.d.ts#L9

**Comment by AdvancedCat** [comment]
题外话：欢迎拥抱 Taro

**Comment by ZakaryCode** [comment]
这个是支持的，提问请备注具体版本哦

> 该问题判断为 QA，将转移到 discussions

**Reply by Tenodec** [reply]
taro版本是3.5.6，使用的react+hooks的模版
我看了view的ViewProps 里面定义为string
我再次尝试了ts报错
`
(property) ViewProps.style?: string | undefined
内联样式。

@supported — alipay

不能将类型“{ marginTop: string; }”分配给类型“string”。
`

**Reply by Tenodec** [reply]
后来我就自己写了个转译函数（CSSProperites To string）临时解决。要是本身支持，是我打开方式对，那就完美～
